tests/test_util/test_warning.py

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python
+
+"""Unit tests for union_dict.
+"""
+from unittest import TestCase, main
+
+from cogent3.util.warning import deprecate
+
+
+__author__ = "Richard Morris"
+__copyright__ = "Copyright 2007-2022, The Cogent Project"
+__credits__ = ["Gavin Huttley", "Richard Morris"]
+__license__ = "BSD-3"
+__version__ = "2023.2.12a1"
+__maintainer__ = "Gavin Huttley"
+__email__ = "gavin.huttley@anu.edu.au"
+__status__ = "Production"
+
+
+class WarningTests(TestCase):
+    """Tests of functions in warnings"""
+
+    def test_deprecate(self):
+        """test that deprecated decorator functions"""
+
+        def new_function():
+            return True
+
+        @deprecate(new_function, "2023.3", "test deprecation")
+        def old_function():
```

Comment:
add a docstring to this function, and add a test that checks the decorated function has the original docstring, e.g.
```python
@deprecate(new_function, "2023.3", "test deprecation")
def old_function():
    """dummy docstring"""
    ...
```
then test as
```python
def test_old_func():
    assert old_func.__doc_ == "dummy docstring"
```

Reply:
You'll need `functools.wraps` for this.